gpkit/tools.py

```diff
@@ -112,3 +115,53 @@ def mdmake(filename, make_tex=True):
         with open(filename+".tex.md", "w") as f:
             f.write(texmd)
     return open(filename+".py")
+
+
+def bound_all_variables(model, eps=1e-30, lower=None, upper=None):
+    "Returns model with additional constraints bounding all free variables"
+    lb = lower if lower else eps
+    ub = upper if upper else 1/eps
+    constraints = []
+    for varkey in model.varkeys:
+        units = varkey.descr.get("units", 1)
+        constraints.append([ub*units >= Variable(**varkey.descr),
+                            Variable(**varkey.descr) >= lb*units])
+    m = Model(model.cost, [constraints, model], model.substitutions)
+    m.bound_all = {"lb": lb, "ub": ub, "varkeys": model.varkeys}
+    return m
+
+
+# pylint: disable=too-many-locals
+def determine_unbounded_variables(model, solver=None, verbosity=0,
+                                  eps=1e-30, lower=None, upper=None, **kwargs):
+    "Returns labeled dictionary of unbounded variables."
+    m = bound_all_variables(model, eps, lower, upper)
+    sol = m.solve(solver, verbosity, **kwargs)
+    lam = sol["sensitivities"]["la"][1:]
+    out = defaultdict(list)
+    for i, varkey in enumerate(m.bound_all["varkeys"]):
+        lam_gt, lam_lt = lam[2*i], lam[2*i+1]
+        if lam_gt and lam_lt:  # to avoid divide-by-zero
+            sens_ratio = lam_gt/lam_lt
+        elif lam_gt:
+            sens_ratio = np.inf
+        else:
+            sens_ratio = 0.0
+        if sens_ratio >= 2:  # arbitrary threshold
+            out["pushing up"].append(varkey)
+        elif sens_ratio <= 0.5:  # arbitrary threshold
+            out["pushing down"].append(varkey)
```

Comment:
Wondering whether comparing the sensitivity ratio is the right thing. Why not just a hard cutoff on sensitivity instead?

(could imagine entries in `lam` floating around in the numerical noise and crossing the ratio threshold, e.g. `1.5e-13` and `2.0e-14`)